pithos/plugins/_mpris.py

```diff
@@ -110,19 +175,40 @@ def _get_playback_status(self):
         else:
             return "Paused"
 
-    def _get_metadata(self):
+    @property
+    def _current_metadata(self):
         """The info for the current song."""
-        return self.__metadata
-
-    def _get_volume(self):
-        return self.window.player.get_property("volume")
+        if self._metadata:
+            return self._metadata
+        else:
+            return {"mpris:trackid": '/org/mpris/MediaPlayer2/TrackList/NoTrack',
+                    #workaround for https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/issues/247
+                    "xesam:url": "",}
+    @property
+    def _current_volume(self):
+        volume = self.window.player.get_property("volume")
+        scaled_volume = math.pow(volume, 1.0/3.0)
```

Comment:
Just piping the volume through to MPRIS results in incorrect volume scaling. I'm not sure why but this fixes that.